doc/api/axes_api.rst

```diff
@@ -1,12 +1,668 @@
-****
-axes
-****
+================
+ ``Axes`` class
+================
+.. currentmodule:: matplotlib.axes
 
+.. autoclass:: Axes
 
-:mod:`matplotlib.axes`
-======================
+.. contents:: Table of Contents
+   :depth: 2
+   :local:
+   :backlinks: entry
 
-.. autoclass:: matplotlib.axes.Axes
-   :members:
-   :undoc-members:
-   :inherited-members:
+
+Plotting
+========
+
+Basic
+-----
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.plot
+   Axes.errorbar
+   Axes.scatter
+
+   Axes.plot_date
+   Axes.step
+
+   Axes.loglog
+   Axes.semilogx
+   Axes.semilogy
+
+   Axes.fill_between
+   Axes.fill_betweenx
+
+   Axes.bar
+   Axes.barh
+
+   Axes.stem
+   Axes.eventplot
+
+   Axes.pie
+
+   Axes.stackplot
+
+
+   Axes.broken_barh
+   Axes.vlines
+   Axes.hlines
+   Axes.fill
+
+Spans
+-----
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.axhline
+   Axes.axhspan
+   Axes.axvline
+   Axes.axvspan
+
+Spectral
+--------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.acorr
+   Axes.angle_spectrum
+   Axes.cohere
+   Axes.csd
+   Axes.magnitude_spectrum
+   Axes.phase_spectrum
+   Axes.psd
+   Axes.specgram
+   Axes.xcorr
+
+Statistics
+----------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.boxplot
+   Axes.violinplot
+
+   Axes.violin
+   Axes.bxp
+
+Binned
+------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.hexbin
+   Axes.hist
+   Axes.hist2d
+
+Contours
+--------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.clabel
+   Axes.contour
+   Axes.contourf
+
+Array
+-----
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.imshow
+   Axes.matshow
+   Axes.pcolor
+   Axes.pcolorfast
+   Axes.pcolormesh
+   Axes.spy
+
+Unstructured Triangles
+----------------------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.tripcolor
+   Axes.triplot
+   Axes.tricontour
+   Axes.tricontourf
+
+
+Text and Annotations
+--------------------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.annotate
+   Axes.text
+   Axes.table
+   Axes.arrow
+
+
+Fields
+------
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.barbs
+   Axes.quiver
+   Axes.quiverkey
+   Axes.streamplot
+
+
+Clearing
+========
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Axes.cla
+   Axes.clear
+
+
+
+Appearance
+==========
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+
+   Axes.axis
+
+   Axes.set_axis_off
+   Axes.set_axis_on
+   Axes.set_frame_on
+   Axes.get_frame_on
+
+   Axes.set_axisbelow
+   Axes.get_axisbelow
+
+   Axes.grid
+
+   Axes.get_axis_bgcolor
```

Comment:
The way this is currently working, these are what is triggering the auto-generate sub-pages to be created.  If we do not include these here then these methods are not documented at all.

Maybe move all the deprecated methods to 'Deprecated' section at the bottom?